src/Google.Cloud.Functions.Invoker.Testing/FunctionTestBase.cs

```diff
@@ -0,0 +1,124 @@
+    /// and disposes of the server when the instance of this class is disposed.
+    /// Additionally, convenience methods to make requests to the server and
+    /// access log entries are provided.
+    /// </summary>
+    public abstract class FunctionTestBase<TFunction> : IDisposable
+    {
+        private static readonly ICloudEventFormatter s_eventFormatter = new JsonEventFormatter();
+        private static readonly ContentType s_applicationJsonContentType = new ContentType("application/json");
+
+        /// <summary>
+        /// The server which will invoke <typeparamref name="TFunction"/>
+        /// when a request is made.
+        /// </summary>
+        public FunctionTestServer<TFunction> Server { get; }
+
+        /// <summary>
+        /// Constructs a new instance using the specified server. Note that
+        /// this effectively takes ownership of the given server, which will be
+        /// disposed when the FunctionTestBase instance is disposed.
+        /// </summary>
+        /// <param name="server">The function test server.</param>
+        protected FunctionTestBase(FunctionTestServer<TFunction> server)
+        {
+            Server = server;
+        }
+
+        /// <summary>
+        /// Constructs a new instance using a new server constructed with default settings.
+        /// </summary>
+        protected FunctionTestBase() : this(new FunctionTestServer<TFunction>())
+        {
+        }
+
+        /// <summary>
+        /// Executes a simple HTTP GET request against an arbitrary URI, and returns
+        /// the text of the response after validating that it indicates success. For
+        /// more complex scenarios, use <see cref="ExecuteHttpRequestAsync(HttpRequestMessage, Action{HttpResponseMessage})"/>.
+        /// </summary>
+        public async Task<string> ExecuteHttpGetRequestAsync()
+        {
+            using var client = Server.CreateClient();
+            using var response = await client.GetAsync("uri");
+            response.EnsureSuccessStatusCode();
+            return await response.Content.ReadAsStringAsync();
+        }
+
+        /// <summary>
+        /// Executes the given HTTP request against the server, calling the validators against the response.
+        /// </summary>
+        public async Task ExecuteHttpRequestAsync(HttpRequestMessage request, Action<HttpResponseMessage> validator)
+        {
+            using var client = Server.CreateClient();
+            using var response = await client.SendAsync(request);
+            validator(response);
+        }
+
+        /// <summary>
+        /// Executes the function with the specified CloudEvent, which is converted
+        /// to an HTTP POST request using structured encoding. This method asserts that the
+        /// request completed successfully.
+        /// </summary>
+        public async Task ExecuteCloudEventRequestAsync(CloudEvent cloudEvent)
```

Comment:
Consider adding an overload taking a validator. We might not need one now, but maybe in the future, and we don't need to release this package just because of that, and also for completeness.